File: src/ayd1ndemirci/EconomyAPI/listener/PlayerListener.php
<?php

namespace ayd1ndemirci\EconomyAPI\listener;

use ayd1ndemirci\EconomyAPI\database\SQLite;
use ayd1ndemirci\EconomyAPI\Economy;
use pocketmine\event\Listener;
use pocketmine\event\player\PlayerJoinEvent;
use SOFe\AwaitGenerator\Await;

class PlayerListener implements Listener
{
    public SQLite $database;

    public function __construct()
    {
        $this->database = Economy::getInstance()->getDatabase();
    }

    public function onPlayerJoin(PlayerJoinEvent $event):void
    {
        $player = $event->getPlayer();
        Await::f2c(function () use($player) {
            $rows = (array) yield from $this->database->getPlayerMoney($player->getName());
            if(empty($rows)) $this->database->addPlayer($player->getName());
        });
    }
}

File: src/main/java/ayd1ndemirci/EconomyAPI/listener/PlayerListener.java
package ayd1ndemirci.EconomyAPI.listener;

import cn.nukkit.Player;
import cn.nukkit.event.EventHandler;
import cn.nukkit.event.Listener;
import cn.nukkit.event.player.PlayerLoginEvent;
import ayd1ndemirci.EconomyAPI.api.EconomyAPI;
import ayd1ndemirci.EconomyAPI.manager.DatabaseManager;

public class PlayerListener implements Listener {

    @EventHandler
    public void onLogin(PlayerLoginEvent event) {
        Player player = event.getPlayer();
        if (!DatabaseManager.existsKey(player.getName())) {
            System.out.println(player.getName() + " doesn't exist");
            EconomyAPI.setMoney(player.getName(), DatabaseManager.getStarterMoney());
        }
    }
}